src/OpenFOAM/matrices/lduMatrix/lduMatrix/lduMatrixATmul.C

```diff
@@ -39,27 +38,27 @@ namespace Foam
 template<bool fast,int nUnroll>
 struct matrixMultiplyFunctor
 {
-    const textures<scalar> psi;
```

Comment:
Why do we get rid of the texturer class? If would be beneficial for older CC.

Reply:
Hi, 
Thanks for your prompt review and helpful feedback. I agree on your points regarding code duplication.

So, if I understand you correctly, the ideal solution would leave lduMatrixATmul, JacobianSmoother, and AINVPreconditioner unchanged, and only involve changes to the textures class defined in textures.H, including runtime checking of the CC?

Best,

Reply:
Yes I think that is the best approach. Also it is probably much easier to implement than modifying all the other classes.